daemon/checkpoint.go

```diff
@@ -38,6 +38,11 @@ func (daemon *Daemon) CheckpointCreate(name string, config types.CheckpointCreat
 		return fmt.Errorf("Invalid checkpoint ID (%s), only %s are allowed", config.CheckpointID, validCheckpointNameChars)
 	}
 
+	checkDir := filepath.Join(daemon.repository, container.ID, "checkpoints", config.CheckpointID)
+	if stat, err := os.Stat(checkDir); err == nil && stat.IsDir() {
+		return fmt.Errorf(" checkpoint with name %s already exists for container %s", config.CheckpointID, container.Name)
+	}
```

Comment:
This doesn't look correct, because a custom `--checkpoint-dir` can be specified, and this will not catch that.

This should probably be something like this;

```patch
diff --git a/daemon/checkpoint.go b/daemon/checkpoint.go
index 33c6001..87006ea 100644
--- a/daemon/checkpoint.go
+++ b/daemon/checkpoint.go
@@ -38,9 +38,8 @@ func (daemon *Daemon) CheckpointCreate(name string, config types.CheckpointCreat
                return fmt.Errorf("Invalid checkpoint ID (%s), only %s are allowed", config.CheckpointID, validCheckpointNameChars)
        }

-       checkDir := filepath.Join(daemon.repository, container.ID, "checkpoints", config.CheckpointID)
-       if stat, err := os.Stat(checkDir); err == nil && stat.IsDir() {
-               return fmt.Errorf(" checkpoint with name %s already exists for container %s", config.CheckpointID, container.Name)
+       if stat, err := os.Stat(filepath.Join(checkpointDir, config.CheckpointID)); err == nil && stat.IsDir() {
+               return fmt.Errorf("Checkpoint with name %s already exists for container %s", config.CheckpointID, container.Name)
        }

        err = daemon.containerd.CreateCheckpoint(container.ID, config.CheckpointID, checkpointDir, config.Exit)
```